code/modules/reagents/chemistry/reagents/cat2_medicine_reagents.dm

```diff
@@ -0,0 +1,212 @@
+	M.adjustOxyLoss(-oxycalc, 0)
+	M.adjustToxLoss(oxycalc/2.5, 0)
+	if(prob(current_cycle) && M.losebreath)
+		M.losebreath--
+	..()
+	return TRUE
+
+/datum/reagent/medicine/C2/perfluorodecalin/overdose_process(mob/living/carbon/human/M)
+	metabolization_rate += 1
+	..()
+	return TRUE
+
+#undef PERF_BASE_DAMAGE
+
+/datum/reagent/medicine/C2/theophylline
+	name = "Theophylline"
+	description = "An oxygen deprivation medication that causes fatigue. Prolonged exposure causes the patient to fall asleep once the medicine metabolizes."
+	color = "#FF6464"
+	var/drowsycd = 0
+
+/datum/reagent/medicine/C2/theophylline/on_mob_life(mob/living/carbon/human/M)
+	M.adjustOxyLoss(-1)
+	M.adjustStaminaLoss(2)
+	if(drowsycd && (world.time > drowsycd))
+		M.drowsyness += 10
+		drowsycd = world.time + (45 SECONDS)
+	else if(!drowsycd)
+		drowsycd = world.time + (15 SECONDS)
+	..()
+	return TRUE
+
+/datum/reagent/medicine/C2/theophylline/on_mob_end_metabolize(mob/living/L)
+	if(current_cycle > 20)
+		L.Sleeping(200)
+	..()
+
+/******TOXIN******/
+
+#define CORTI_WEAKRATE	0.005
+
+/datum/reagent/medicine/C2/corticoline //made up chem from Corticosteriods, used to treat POISON ivy wink wink
+  name = "Corticoline"
+  description = "An antitoxin that temporarily weakens the user, making them susceptible to other forms of damage. Weakness and toxin healing scales with exposure."
+  overdose_threshold = 11 // Scales every damage so naturally quite abusable. Luckily the OD purges the chem which resets the damage mod *wink*
+  var/reset_mods = 0 //amount, not bool
+
+/datum/reagent/medicine/C2/corticoline/on_mob_life(mob/living/carbon/human/M)
+	var/maths = CORTI_WEAKRATE * current_cycle
+	reset_mods += maths
+	var/datum/physiology/phis = M.physiology
+	phis.brute_mod += maths
+	phis.burn_mod += maths
+	phis.oxy_mod += maths
+	phis.stamina_mod += maths
+	M.adjustToxLoss(round(reset_mods*-1000,0.01)) //Math is fun!
+	..()
+	return TRUE
+
+/datum/reagent/medicine/C2/corticoline/on_mob_delete(mob/living/carbon/human/M)
+	var/datum/physiology/phis = M.physiology
+	phis.brute_mod = max(phis.brute_mod - reset_mods, 1)
+	phis.burn_mod = max(phis.burn_mod - reset_mods, 1)
+	phis.oxy_mod = max(phis.oxy_mod - reset_mods, 1)
+	return ..()
+
+/datum/reagent/medicine/C2/corticoline/overdose_process(mob/living/carbon/human/M)
+	var/maths = reset_mods*3
+	M.adjustBruteLoss(maths)
+	M.adjustFireLoss(maths)
+	M.adjustOxyLoss(maths)
+	..()
+	M.reagents.del_reagent(type)
+	return TRUE
+
+#undef CORTI_WEAKRATE
+
+/datum/reagent/medicine/C2/palletta //made up chem, it's a PALLETTE cleanser hehe
+	name = "Palletta"
+	description = "An antitoxin that scales with the more chems in the body as well as purges chems (including itself). Causes loss of breath."
+
+/datum/reagent/medicine/C2/palletta/on_mob_life(mob/living/carbon/human/M)
+	var/total_v = 0
+	for(var/r in M.reagents.reagent_list)
+		var/datum/reagent/the_reagent = r
+		total_v += the_reagent.volume
+	M.adjustToxLoss(total_v*-0.1)
```

Comment:
Chug water for ten seconds straight, pop a pill, instantly cure all toxin damage. Factor in perfluorodecalin and you're immune to both oxyloss and toxin for the most part.

Reply:
I forgot to calculate for chem variety as much as volume